lib/auth/github.go

```diff
@@ -537,7 +537,7 @@ func newGithubOAuth2Config(connector types.GithubConnector) oauth2.Config {
 
 // ValidateGithubAuthCallback validates Github auth callback redirect
 func (a *Server) validateGithubAuthCallback(ctx context.Context, diagCtx *SSODiagContext, q url.Values) (*authclient.GithubAuthResponse, error) {
-	logger := log.WithFields(logrus.Fields{teleport.ComponentKey: "github"})
+	logger := a.logger.With(teleport.ComponentKey, "github")
```

Comment:
I feel like `validateGithubAuthCallback` and friends could benefit from refactoring. Right now is probably not the best time, but the original code didn't anticipate the use-cases that have been added with time and it shows. Possibly the opportunity for this will arrive in the future?